xml_analyzer/calculator.py

```diff
@@ -1,63 +1,26 @@
-import xml.etree.ElementTree as Et
-from enum import Enum
-from .formula import Formula
 from .data_container import DataContainer
-
-
-class EquationTypes(Enum):
-    equation = 'equation'
-    inequation_gr_th = 'inequation_gr_th'
-    inequation_sm_th = 'inequation_sm_th'
-    inequation_gr_eq = 'inequation_gr_eq'
-    inequation_sm_eq = 'inequation_sm_eq'
+from .formula import Formula
 
 
 class Calculator:
-    def __init__(self, constraint_property: Et.Element, property_value_mapping: DataContainer):
-        self.constraint_property = constraint_property
-        self.property_value_mapping = property_value_mapping
+    def __init__(self, data: DataContainer):
+        self._data = data
+        self._formula = Formula(data.constraint_specification, data.get_variable_names())
 
     @staticmethod
-    def get_expression_type(expr: str) -> [EquationTypes, str]:
-
-        if '>=' in expr:
-            expr_type = EquationTypes.inequation_gr_eq
-            expr_sep = '>='
-        elif '<=' in expr:
-            expr_type = EquationTypes.inequation_sm_eq
-            expr_sep = '<='
-        elif '>' in expr:
-            expr_type = EquationTypes.inequation_gr_th
-            expr_sep = '>'
-        elif '<' in expr:
-            expr_type = EquationTypes.inequation_sm_th
-            expr_sep = '<'
-        elif '=' in expr:
-            expr_type = EquationTypes.equation
-            expr_sep = '='
-        else:
-            expr_type = 'undef'
-            expr_sep = 'undef'
-
-        return expr_type, expr_sep
-
-    def resolve_auto_calc_function(self, autocalc_method, type_name):
-        retval = None
+    def _resolve_auto_calc_functions(autocalc_method, values) -> float:
         if autocalc_method == 'autosum':
-            retval = self.do_autosum(type_name)
-
-        if autocalc_method == 'automult':
-            pass
+            return round(sum(values), 3)
+        return None
 
-        return retval
+    def calculate(self) -> float:
+        for autocalc_method in self._data.autocalc_mappings:
+            for autocalc_entry in self._data.autocalc_mappings[autocalc_method]:
+                sysml_type_name = autocalc_entry.property
+                auto_value = self._resolve_auto_calc_functions(autocalc_method, autocalc_entry.value)
```

Comment:
Return value None is not tested

Reply:
Fixed with https://github.com/Squiddrone/VPParametricsSolver/pull/2/commits/7ff4099ad757bf5d465260c8ac8cb42df12253f0